l10n_br_resource/models/resource_calendar.py

```diff
@@ -0,0 +1,303 @@
+                                    calendario corrente
+                                    date_start <= data_referencia <= data_end
+
+        :return boolean True se a data referencia for feriado
+                        False se a data referencia nao for feriado
+        """
+        if not self.leave_ids:
+            self = self.with_context(self._context)._get_brazilian_calendar()
+        for leave in self.leave_ids:
+            if leave.date_from <= data_referencia.strftime(
+                    "%Y-%m-%d %H:%M:%S"):
+                if leave.date_to >= data_referencia. \
+                        strftime("%Y-%m-%d %H:%M:%S"):
+                    if leave.leave_kind in ('F', 'B'):
+                        return True
+        return False
+
+    @api.multi
+    def data_eh_dia_util(self, data=datetime.now()):
+        """Verificar se data é dia util.
+        :param datetime data: Se nenhuma data referencia for passada
+                              verifique o dia de hoje.
+        :return boolean True: Se for dia útil
+                        False: Se Não for dia útil
+        """
+        return not self.data_eh_feriado(data) and data.weekday() <= 4 or False
+
+    @api.multi
+    def data_eh_dia_util_bancario(self, data=datetime.now()):
+        """Verificar se data é dia util.
+        :param datetime data: Se nenhuma data referencia for passada
+                              verifique o dia de hoje.
+        :return boolean True: Se for dia útil
+                        False: Se Não for dia útil
+        """
+        if data.weekday() > 4:
+            return False
+        elif self.data_eh_feriado_bancario(data):
+            return False
+        return True
+
+    @api.multi
+    def quantidade_dias_uteis(
+            self, data_inicio=datetime.now(), data_fim=datetime.now()):
+        """Calcular a quantidade de dias úteis em determinado período.
+        :param datetime data_inicio: Se nenhuma data referencia for passada
+                                   verifique o dia de hoje.
+        :param datetime data_fim: Se nenhuma data referencia for passada
+                                   verifique o dia de hoje.
+        :return int: Quantidade de dias úteis
+        """
+        dias_uteis = 0
+        while data_inicio <= data_fim:
+            if self.data_eh_dia_util(data_inicio):
+                dias_uteis += 1
+            data_inicio += timedelta(days=1)
+
+        return dias_uteis
+
+    @api.multi
+    def proximo_dia_util(self, data_referencia=datetime.now()):
+        """Retornar o próximo dia util.
+        :param datetime data_referencia: Se nenhuma data referencia for passada
+                                   verifique se amanha é dia útil.
+        :return datetime Proximo dia util apartir da data referencia
+        """
+        while data_referencia:
+            if self.data_eh_dia_util(data_referencia):
+                return data_referencia
+            data_referencia += timedelta(days=1)
```

Comment:
There is a problem here, the function return the next useful day ( proximo_dia_util ), but the first check is with Date receive, what makes wrong result, because if the date receive is useful day the function return the same date instead of the next useful day, to solve the simple way is just put "data_referencia += timedelta(days=1)" before the "if" . I saw the error by checking the tests, you can see it happening on test "def test_06_obter_proximo_dia_util(self)".